diff --git a/encoder/3rdparty/tinydds.h b/encoder/3rdparty/tinydds.h
index 6b4712a..01a42d6 100644
--- a/encoder/3rdparty/tinydds.h
+++ b/encoder/3rdparty/tinydds.h
@@ -1379,14 +1379,14 @@ bool TinyDDS_ReadHeader(TinyDDS_ContextHandle handle) {
 					ctx->header.mipMapCount = i + 1;
 					break;
 				}
-			} else if (w <= 1 || h <= 1) {
+			} else if (w <= 1 && h <= 1) {
 				ctx->header.mipMapCount = i + 1;
 				break;
 			}
 
 
-			w = w / 2;
-			h = h / 2;
+			w = w > 1 ? w / 2 : 1;
+			h = h > 1 ? h / 2 : 1;
 		}
 
 	}
diff --git a/encoder/basisu_comp.cpp b/encoder/basisu_comp.cpp
index dee91aa..d76b2a8 100644
--- a/encoder/basisu_comp.cpp
+++ b/encoder/basisu_comp.cpp
@@ -1481,6 +1481,7 @@ namespace basisu
 
 		m_uastc_backend_output.m_tex_format = basist::basis_tex_format::cUASTC_LDR_4x4;
 		m_uastc_backend_output.m_etc1s = false;
+		m_uastc_backend_output.m_srgb = m_params.m_ktx2_and_basis_srgb_transfer_function;
 		m_uastc_backend_output.m_slice_desc = m_slice_descs;
 		m_uastc_backend_output.m_slice_image_data.resize(m_slice_descs.size());
 		m_uastc_backend_output.m_slice_image_crcs.resize(m_slice_descs.size());
